pkg/compiler/compiler.go

```diff
@@ -75,3 +89,18 @@ func (c *Compiler) compileTests(binaryName, path string) error {
 
 	return cmd.Run()
 }
+
+func executebleExists(path string) bool {
+	fi, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+
+	if fi.IsDir() {
+		return false
+	}
+
+	isExecutable := fi.Mode()&os.FileMode(0111) != 0
```

Comment:
That looks super magic to me. Some information on what is going on here would be nice.

Reply:
Sure, added some comments, PTAL.